app/app/src/main/kotlin/com/hedvig/android/app/ui/HedvigAppState.kt

```diff
@@ -190,6 +188,17 @@ internal class HedvigAppState(
     }
 }
 
+@JvmInline
+value class NavigationSuiteType private constructor(private val description: String) {
```

Comment:
Oh cool, didn't know about value class :) Have read about them now, handy
One question: why override String here? can't find usages. In other words, why do we need description?

Reply:
Yeah you are right this looks a bit odd.
It ended up being like this as I used the code from here https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/material3/material3-adaptive-navigation-suite/src/commonMain/kotlin/androidx/compose/material3/adaptive/navigationsuite/NavigationSuiteScaffold.kt;l=336-374?q=NavigationSuiteType and how they do it in the latest alphas of material 3 (it's experimental there atm) so I just kept it the same way. I started from their impl and stripped away stuff until we got what we had before too, only in a nice `NavigationSuite` composable.

The reason they do it this way from my understanding is so that they are able to add more cases in the future, without breaking any older callers. Which a sealed interface would not handle. While also not adding any overhead with an extra class, so they just go with a value class, which practically when the app runs this will look like a `String` in the code at runtime.
The description therefore is used exactly for that, so that after the value class is gone, at runtime these three cases will just be seen as something like "NavigationBar", "NavigationRail" and "None".

To be fair we could absolutely not do all this to get whatever minuscule benefits it gives us, but it's fine this way too I think :D